### PR TITLE
[14.0][FIX] shopinvader_invoice: Add set_session in get()

### PR DESCRIPTION
**Pull request by rousseldenis**
In order to get correct header filled in with an entry
HTTP_SESS_INVOICE_ID, add the set_session with the invoice id.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@acsonefho @Cedric-Pigeon

**Review by acsonefho — APPROVED** [review]

File: shopinvader_invoice/services/invoice.py [hunk 1]
```diff
@@ -13,7 +13,10 @@ def get(self, _id):
         :return: dict/json
         """
         invoice = self._get(_id)
-        result = {"data": self._to_json(invoice)[0]}
+        result = {
+            "data": self._to_json(invoice)[0],
+            "set_session": {"invoice_id": invoice.id},
```
Comment (simahawk) [inline shopinvader_invoice/services/invoice.py:18]: curious: what's the goal for having the invoice_id in the session?

Comment (rousseldenis) [reply shopinvader_invoice/services/invoice.py:18]: > curious: what's the goal for having the invoice_id in the session?

The idea is to mimic the cart behaviour.

A place where this is mandatory is in payments. As this is the only way to know if we are in a shopinvader behaviour, we need to add it in headers if a service is exposed publicly as webhooks.

See the behaviour for cart payments:

https://github.com/shopinvader/odoo-shopinvader-payment/blob/14.0/shopinvader_payment/components/payment_transaction_event_listerner.py#L26

**Review by sbidoul — CHANGES_REQUESTED** [review]
So yeah, as discussed, let's not do this. The front-end dev need to pass the invoice id around themselves. 

We need to kill this whole store_cache / session concept, so let's not add new uses for it.

**Pull request closed without merging**